src/Sylius/Bundle/ResourceBundle/Controller/ResourceController.php

```diff
@@ -316,11 +325,18 @@ public function updateAction(Request $request)
                 return $this->redirectHandler->redirectToResource($configuration, $resource);
             }
 
-            if ($configuration->hasStateMachine()) {
-                $this->stateMachine->apply($configuration, $resource);
+            try {
+                $this->resourceUpdater->applyTransitionAndFlush($resource, $configuration, $this->manager);
+            } catch (RaceConditionException $exception) {
+                if (!$configuration->isHtmlRequest()) {
+                    return $this->viewHandler->handle($configuration, View::create($form, Response::HTTP_BAD_REQUEST));
```

Comment:
What do you think about [423 Locked](https://tools.ietf.org/html/rfc4918#page-17)? Anyway, this makes sense only if we will stay with `RaceConditionException`. If we decide for something more generic, like `ResourceException` we will have to decide for something more generic.